quickwit/quickwit-serve/src/lib.rs

```diff
@@ -709,34 +735,43 @@ async fn setup_ingest_v2(
         )
         .await?;
         ingester.subscribe(event_broker);
-        let ingester_service = IngesterServiceClient::new(ingester);
-        Some(ingester_service)
+        Some(ingester)
     } else {
         None
     };
     // Setup ingester pool change stream.
-    let ingester_service_opt_clone = ingester_service_opt.clone();
+    let ingester_opt_clone = ingester_opt.clone();
     let cluster_change_stream = cluster.ready_nodes_change_stream().await;
     let max_message_size = node_config.grpc_config.max_message_size;
     let ingester_change_stream = cluster_change_stream.filter_map(move |cluster_change| {
-        let ingester_service_opt = ingester_service_opt_clone.clone();
+        let ingester_opt_clone_clone = ingester_opt_clone.clone();
         Box::pin(async move {
             match cluster_change {
                 ClusterChange::Add(node)
                     if node.enabled_services().contains(&QuickwitService::Indexer) =>
                 {
                     let node_id: NodeId = node.node_id().into();
 
+                    // Remember that movie called "Inception"?
                     if node.is_self_node() {
-                        let ingester_service =
-                            ingester_service_opt.expect("ingester service should be initialized");
+                        let ingester = ingester_opt_clone_clone
+                            .expect("ingester service should be initialized");
+                        let layers = ServiceBuilder::new()
+                            .layer(INGEST_GRPC_CLIENT_METRICS_LAYER.clone())
+                            .layer(INGEST_GRPC_SERVER_METRICS_LAYER.clone())
```

Comment:
do we want both here?

Reply:
I updated the PR with the following comment:
> // Here, since the service is available locally, we bypass the gRPC layer
                        // and use the mailbox directly. However, we still want client-side metrics,
                        // so we use both metrics layers.